x/nexus/keeper/chain.go

```diff
@@ -62,6 +63,29 @@ func (k Keeper) IsAssetRegistered(ctx sdk.Context, chain exported.Chain, denom s
 	return chainState.HasAsset(denom)
 }
 
+func (k Keeper) setFeeInfo(ctx sdk.Context, chain exported.Chain, asset string, feeInfo exported.FeeInfo) {
+	k.getStore(ctx).Set(assetFeePrefix.Append(utils.LowerCaseKey(chain.Name)).Append(utils.KeyFromStr(asset)), &feeInfo)
+}
+
+// GetFeeInfo retrieves the fee info for an asset on a chain, and returns zero fees if it doesn't exist
+func (k Keeper) GetFeeInfo(ctx sdk.Context, chain exported.Chain, asset string) (feeInfo exported.FeeInfo, found bool) {
+	feeInfo = exported.ZeroFee()
+	return feeInfo, k.getStore(ctx).Get(assetFeePrefix.Append(utils.LowerCaseKey(chain.Name)).Append(utils.KeyFromStr(asset)), &feeInfo)
+}
+
+// RegisterFee registers the fee info for an asset on a chain
+func (k Keeper) RegisterFee(ctx sdk.Context, chain exported.Chain, asset string, feeInfo exported.FeeInfo) error {
+	if !k.IsAssetRegistered(ctx, chain, asset) {
+		return fmt.Errorf("%s is not a registered asset for chain %s", asset, chain.Name)
+	}
+
+	k.setFeeInfo(ctx, chain, asset, feeInfo)
+
+	ctx.Logger().Info(fmt.Sprintf("registered fee info for asset %s on chain %s", asset, chain.Name), types.AttributeKeyChain, chain.Name, types.AttributeKeyAsset, asset)
```

Comment:
It's preferred not to have logging in the keepers. Also, use `k.Logger(ctx)` instead of `ctx.Logger()`.